x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -724,8 +742,11 @@ public Map<String, Processor.Factory> getProcessors(Processor.Parameters paramet
     private final SetOnce<DeploymentManager> deploymentManager = new SetOnce<>();
     private final SetOnce<TrainedModelAssignmentClusterService> trainedModelAllocationClusterServiceSetOnce = new SetOnce<>();
 
+    private final List<MachineLearningExtension> machineLearningExtensions;
```

Comment:
Instead of having a list, I think it would be better to have a `SetOnce`. We don't expect multiple extensions and having more than 1 would create a problem about how to combine their output.